siliconcompiler/tools/vivado/compile.tcl

```diff
@@ -18,12 +18,14 @@ if {[dict exists $sc_cfg input fpga xdc]} {
 set sc_partname   [dict get $sc_cfg fpga partname]
 set sc_step       [dict get $sc_cfg arg step]
 set sc_index      [dict get $sc_cfg arg index]
+set sc_flow       [dict get $sc_cfg option flow]
+set sc_task       [dict get $sc_cfg flowgraph $sc_flow $sc_step $sc_index task]
 
 ##############################
 # Flow control
 ##############################
 
-if {$sc_step == "syn"} {
+if {$sc_task == "syn_fpga"} {
```

Comment:
how do we get here? The task is compile, so we can't be checking for syn_fpga, is this check redundant?

Reply:
Good question - I don't know why there is a check for the synthesis step in the Vivado `compile` script.

It looks like there is synthesis logic contained in the block, so I don't want to delete it if it might be used somewhere. @nmoroze ?

Reply:
The Vivado flow is currently implemented as a single file that implements every step/task using a branch:

```tcl
# common setup...

if {$sc_step == "syn"} {
  # syn logic
 } elseif ($sc_step == "place"} {
  # place logic
} # etc...

# common finish...
```

so this branch needs to stay, and we do want to make this change. The other checks in this file should change to be based on `$sc_task` as well, but I can address this in a follow-up PR.